src/Query/ParserResult.php

```diff
@@ -87,13 +97,38 @@ public function setSqlExecutor($executor)
     /**
      * Gets the SQL executor used by this ParserResult.
      *
-     * @return AbstractSqlExecutor
+     * @return ?AbstractSqlExecutor
      */
     public function getSqlExecutor()
     {
         return $this->sqlExecutor;
     }
 
+    public function setSqlFinalizer(SqlFinalizer $finalizer): void
+    {
+        $this->sqlFinalizer = $finalizer;
+    }
+
+    public function getSqlFinalizer(): SqlFinalizer
+    {
+        if ($this->sqlFinalizer === null) {
+            throw new RuntimeException('This ParserResult was not created with an SqlFinalizer');
```

Comment:
Is `RuntimeException` the right type of exception for this? I don't think it falls into the category of "checked exception" (see https://phpstan.org/blog/bring-your-exceptions-under-control)